### PR TITLE
add capturing to Start/Resume

### PR DESCRIPTION
**Pull request by sinkingsugar**

### CODE REVIEW

**Comment by sinkingsugar** [comment]
@guusw please review 🙏 as I need this for discord bot

File: src/core/shards/wires.cpp [hunk 1]
```diff
@@ -469,26 +536,44 @@ struct Resume : public WireBase {
     }();
 
     // assign the new wire as current wire on the flow
-    context->flow->wire = pwire;
+    context->flow->wire = p_wire;
```
Comment (guusw) [inline src/core/shards/wires.cpp:539]: Stick with camelCase -> pWire?

File: src/core/shards/wires.cpp [hunk 2]
```diff
@@ -469,26 +536,44 @@ struct Resume : public WireBase {
...
-    if (shards::hasEnded(pwire)) {
-      shards::stop(pwire);
+    if (shards::hasEnded(p_wire)) {
+      shards::stop(p_wire);
+    }
+
+    // capture variables
+    for (auto &v : _vars) {
+      auto &var = v.get();
+      auto &v_ref = p_wire->variables[v.variableName()];
```
Comment (guusw) [inline src/core/shards/wires.cpp:549]: camelCase

File: src/core/shards/wires.cpp [hunk 3]
```diff
@@ -510,9 +600,10 @@ struct Start : public Resume {
   SHVar activate(SHContext *context, const SHVar &input) {
     auto current = context->wireStack.back();
 
-    auto pwire = [&] {
+    auto p_wire = [&] {
```
Comment (guusw) [inline src/core/shards/wires.cpp:603]: camelCase

**Review by guusw — COMMENTED** [review]
Ah forgot to review earlier, only had some style fixes.

**Pull request merged**